server/performance/metrics.go

```diff
@@ -81,53 +80,35 @@ func NewMetrics() *Metrics {
 	)
 	m.registry.MustRegister(m.StoreOpCounters)
 
-	m.RTPPacketCounters = prometheus.NewCounterVec(
-		prometheus.CounterOpts{
-			Namespace: metricsNamespace,
-			Subsystem: metricsSubSystemRTC,
-			Name:      "rtp_packets_total",
-			Help:      "Total number of sent/received RTP packets",
-		},
-		[]string{"direction", "type"},
-	)
-	m.registry.MustRegister(m.RTPPacketCounters)
-
-	m.RTPPacketBytesCounters = prometheus.NewCounterVec(
-		prometheus.CounterOpts{
-			Namespace: metricsNamespace,
-			Subsystem: metricsSubSystemRTC,
-			Name:      "rtp_bytes_total",
-			Help:      "Total number of sent/received RTP packet bytes",
-		},
-		[]string{"direction", "type"},
-	)
-	m.registry.MustRegister(m.RTPPacketBytesCounters)
-
-	m.RTCSessions = prometheus.NewGaugeVec(
-		prometheus.GaugeOpts{
-			Namespace: metricsNamespace,
-			Subsystem: metricsSubSystemRTC,
-			Name:      "sessions_total",
-			Help:      "Total number of active RTC sessions",
-		},
-		[]string{"channelID"},
-	)
-	m.registry.MustRegister(m.RTCSessions)
-
-	m.RTCConnStateCounters = prometheus.NewCounterVec(
-		prometheus.CounterOpts{
-			Namespace: metricsNamespace,
-			Subsystem: metricsSubSystemRTC,
-			Name:      "conn_states_total",
-			Help:      "Total number of RTC connection state changes",
-		},
-		[]string{"type"},
-	)
-	m.registry.MustRegister(m.RTCConnStateCounters)
+	m.rtcMetrics = perf.NewMetrics(metricsNamespace, m.registry)
 
 	return &m
 }
 
+func (m *Metrics) RTCMetrics() rtc.Metrics {
+	return m.rtcMetrics
+}
+
 func (m *Metrics) Handler() http.Handler {
 	return promhttp.HandlerFor(m.registry, promhttp.HandlerOpts{})
 }
+
+func (m *Metrics) IncWebSocketEvent(direction, evType string) {
+	m.WebSocketEventCounters.With(prometheus.Labels{"direction": direction, "type": evType}).Inc()
+}
```

Comment:
Thought: could most of these be direct calls to an initialized struct from the rtcd package?

Reply:
Not sure I understood the concern but we are passing the registry during initialization:

https://github.com/mattermost/mattermost-plugin-calls/blob/4b570fa53f800c0c0d2841f44ce5883d7cf01556/server/performance/metrics.go#L83

So what's really happening is that we are just extending the existing metrics with the rtc related counters. Anything on this side is called on the `Metrics` structure defined by the plugin.